lib/cli/src/Cardano/CLI.hs

```diff
@@ -604,11 +604,19 @@ execLaunch
 execLaunch verbosity stateDir withStateDir commands = do
     installSignalHandlers
     (_, _, tracer) <- initTracer (verbosityToMinSeverity verbosity) "launch"
-    setupStateDir (logInfo tracer) (withStateDir tracer) stateDir
-    logInfo tracer $ fmt $ nameF "launch" $ blockListF commands
-    (ProcessHasExited pName code) <- launch commands
-    logAlert tracer $ T.pack pName <> " exited with code " <> T.pack (show code)
-    exitWith code
+    setupStateDir (logInfo tracer) (withStateDir tracer) stateDir >>= \case
+        Left _ -> do
+            putErrLn $ mconcat
```

Comment:
Use `logError tracer` rather that `putErrLn`.

Reply:
Hm, I actually thought that `putErrLn` fits better here...
Similar case as -> https://github.com/input-output-hk/cardano-wallet/blob/master/lib/cli/src/Cardano/CLI.hs#L1283

Reply:
I'd indeed stick with `putErrLn` for failures of the command-line, and keep the logger for failures of the application / server.

Reply:
That's arguable however... since we use the logger pretty much everywhere else in this function...

Reply:
:confused: